benchmarks/mpas_ocean.py

```diff
@@ -164,3 +164,29 @@ def time_nearest_neighbor_remapping(self):
 
     def time_inverse_distance_weighted_remapping(self):
         self.uxds_480["bottomDepth"].remap.inverse_distance_weighted(self.uxds_120.uxgrid)
+
+from uxarray.grid.coordinates import _construct_face_centerpoints, _construct_face_centroids
```

Comment:
This line needs to be moved either within the `class ConstructFaceLatLon` or inside of the `time_welzl`  method for the benchmarks to run correctly.

Reply:
> This line needs to be moved either within the `class ConstructFaceLatLon` or inside of the `time_welzl` method for the benchmarks to run correctly.

All benchmarks run to completion:
`asv run --bench mpas_ocean.ConstructTreeStructures.time_kd_tree     --python=same`, 

the problem is profiling, eg:

`asv profile mpas_ocean.ConstructTreeStructures.time_kd_tree` 
doesn't work, something to do with passing arguments `resolution etc.` keeps getting error  
`mpas_ocean.ConstructTreeStructures.time_kd_tree benchmark not found`

Reply:
> ## ASV Benchmarking
> Benchmark Comparison Results
> Benchmarks that have improved:
> 
> Change	Before [[82e46b8](https://github.com/UXARRAY/uxarray/commit/82e46b8c79bb0107753398ddab096c5f6eb9b718)]	After [[9f952f6](https://github.com/UXARRAY/uxarray/commit/9f952f6fff44f428c382d0c51bc64d6ebd9cf536)]	Ratio	Benchmark (Parameter)
> failed	1.51±0.01s	n/a	mpas_ocean.ConstructFaceLatLon.time_welzl('120km')
> failed	95.5±0.1ms	n/a	mpas_ocean.ConstructFaceLatLon.time_welzl('480km')
> -	408M	354M	0.87	mpas_ocean.Integrate.peakmem_integrate('480km')
> Benchmarks that have stayed the same:
> 
> Change	Before [[82e46b8](https://github.com/UXARRAY/uxarray/commit/82e46b8c79bb0107753398ddab096c5f6eb9b718)]	After [[9f952f6](https://github.com/UXARRAY/uxarray/commit/9f952f6fff44f428c382d0c51bc64d6ebd9cf536)]	Ratio	Benchmark (Parameter)
> 379M	380M	1.00	face_bounds.FaceBounds.peakmem_face_bounds(PosixPath('/home/runner/work/uxarray/uxarray/test/meshfiles/mpas/QU/oQU480.231010.nc'))
> 380M	381M	1.00	face_bounds.FaceBounds.peakmem_face_bounds(PosixPath('/home/runner/work/uxarray/uxarray/test/meshfiles/scrip/outCSne8/outCSne8.nc'))
> 383M	384M	1.00	face_bounds.FaceBounds.peakmem_face_bounds(PosixPath('/home/runner/work/uxarray/uxarray/test/meshfiles/ugrid/geoflow-small/grid.nc'))
> 381M	382M	1.00	face_bounds.FaceBounds.peakmem_face_bounds(PosixPath('/home/runner/work/uxarray/uxarray/test/meshfiles/ugrid/quad-hexagon/grid.nc'))
> 1.36±0.02s	1.35±0s	1.00	face_bounds.FaceBounds.time_face_bounds(PosixPath('/home/runner/work/uxarray/uxarray/test/meshfiles/mpas/QU/oQU480.231010.nc'))
> 176±3ms	177±1ms	1.01	face_bounds.FaceBounds.time_face_bounds(PosixPath('/home/runner/work/uxarray/uxarray/test/meshfiles/scrip/outCSne8/outCSne8.nc'))
> 1.64±0.01s	1.65±0.02s	1.00	face_bounds.FaceBounds.time_face_bounds(PosixPath('/home/runner/work/uxarray/uxarray/test/meshfiles/ugrid/geoflow-small/grid.nc'))
> 7.63±0.08ms	7.83±0.2ms	1.03	face_bounds.FaceBounds.time_face_bounds(PosixPath('/home/runner/work/uxarray/uxarray/test/meshfiles/ugrid/quad-hexagon/grid.nc'))
> 1.68±0.01s	1.70±0.01s	1.01	import.Imports.timeraw_import_uxarray
> 621±4ms	623±4ms	1.00	mpas_ocean.ConnectivityConstruction.time_face_face_connectivity('120km')
> 39.3±0.3ms	39.6±0.4ms	1.01	mpas_ocean.ConnectivityConstruction.time_face_face_connectivity('480km')
> 1.80±0.02ms	1.98±0.2ms	~1.10	mpas_ocean.ConnectivityConstruction.time_n_nodes_per_face('120km')
> 474±7μs	476±7μs	1.00	mpas_ocean.ConnectivityConstruction.time_n_nodes_per_face('480km')
> 361±7ms	356±3ms	0.99	mpas_ocean.ConstructFaceLatLon.time_cartesian_averaging('120km')
> 23.4±0.3ms	22.9±0.2ms	0.98	mpas_ocean.ConstructFaceLatLon.time_cartesian_averaging('480km')
> 1.19±0μs	1.23±0.01μs	1.04	mpas_ocean.ConstructTreeStructures.time_ball_tree('120km')
> 305±5ns	303±4ns	1.00	mpas_ocean.ConstructTreeStructures.time_ball_tree('480km')
> 794±5ns	873±1ns	1.10	mpas_ocean.ConstructTreeStructures.time_kd_tree('120km')
> 280±2ns	291±4ns	1.04	mpas_ocean.ConstructTreeStructures.time_kd_tree('480km')
> 402M	399M	0.99	mpas_ocean.GeoDataFrame.peakmem_to_geodataframe('120km', False)
> 390M	387M	0.99	mpas_ocean.GeoDataFrame.peakmem_to_geodataframe('120km', True)
> 362M	362M	1.00	mpas_ocean.GeoDataFrame.peakmem_to_geodataframe('480km', False)
> 360M	361M	1.00	mpas_ocean.GeoDataFrame.peakmem_to_geodataframe('480km', True)
> 1.06±0s	1.04±0.01s	0.98	mpas_ocean.GeoDataFrame.time_to_geodataframe('120km', False)
> 57.4±0.5ms	56.9±0.3ms	0.99	mpas_ocean.GeoDataFrame.time_to_geodataframe('120km', True)
> 77.7±0.8ms	77.3±1ms	1.00	mpas_ocean.GeoDataFrame.time_to_geodataframe('480km', False)
> 5.53±0.2ms	5.22±0.09ms	0.94	mpas_ocean.GeoDataFrame.time_to_geodataframe('480km', True)
> 263M	263M	1.00	mpas_ocean.Gradient.peakmem_gradient('120km')
> 240M	240M	1.00	mpas_ocean.Gradient.peakmem_gradient('480km')
> 2.66±0.01ms	2.74±0.05ms	1.03	mpas_ocean.Gradient.time_gradient('120km')
> 280±0.6μs	284±6μs	1.01	mpas_ocean.Gradient.time_gradient('480km')
> 370M	370M	1.00	mpas_ocean.Integrate.peakmem_integrate('120km')
> 175±1ms	176±0.6ms	1.00	mpas_ocean.Integrate.time_integrate('120km')
> 11.7±0.03ms	12.2±0.04ms	1.04	mpas_ocean.Integrate.time_integrate('480km')
> 343±2ms	343±2ms	1.00	mpas_ocean.MatplotlibConversion.time_dataarray_to_polycollection('120km', 'exclude')
> 346±2ms	343±2ms	0.99	mpas_ocean.MatplotlibConversion.time_dataarray_to_polycollection('120km', 'include')
> 346±3ms	342±2ms	0.99	mpas_ocean.MatplotlibConversion.time_dataarray_to_polycollection('120km', 'split')
> 21.6±0.3ms	22.3±0.6ms	1.03	mpas_ocean.MatplotlibConversion.time_dataarray_to_polycollection('480km', 'exclude')
> 21.6±0.1ms	21.5±0.2ms	1.00	mpas_ocean.MatplotlibConversion.time_dataarray_to_polycollection('480km', 'include')
> 21.4±0.1ms	21.6±0.1ms	1.01	mpas_ocean.MatplotlibConversion.time_dataarray_to_polycollection('480km', 'split')
> 54.7±0.2ms	54.8±0.2ms	1.00	mpas_ocean.RemapDownsample.time_inverse_distance_weighted_remapping
> 44.2±0.03ms	44.4±0.2ms	1.00	mpas_ocean.RemapDownsample.time_nearest_neighbor_remapping
> 361±0.8ms	361±1ms	1.00	mpas_ocean.RemapUpsample.time_inverse_distance_weighted_remapping
> 263±0.3ms	263±1ms	1.00	mpas_ocean.RemapUpsample.time_nearest_neighbor_remapping
> failed	failed	n/a	quad_hexagon.ConstructFaceLatLon.time_welzl
> 240M	237M	0.99	quad_hexagon.QuadHexagon.peakmem_open_dataset
> 236M	236M	1.00	quad_hexagon.QuadHexagon.peakmem_open_grid
> 6.35±0.1ms	6.56±0.2ms	1.03	quad_hexagon.QuadHexagon.time_open_dataset
> 5.43±0.07ms	5.50±0.02ms	1.01	quad_hexagon.QuadHexagon.time_open_grid
> 236M	236M	1.00	quad_hexagon.QuadHexagon2.peakmem_open_grid2
> 6.29±0.03ms	6.51±0.1ms	1.03	quad_hexagon.QuadHexagon2.time_open_dataset
> 5.39±0.04ms	5.58±0.02ms	1.03	quad_hexagon.QuadHexagon2.time_open_grid

@rajeeja 

The results look correct now (compared to what we were getting before). Does running `asv profile` still occur for you locally?

Reply:
Yes, bench works. profile doesn't - any function with arguments in mpas_ocean file